### PR TITLE
Gives the ability to adjust another persons suit sensors.

### LINKED ISSUE #299: Adjust Suit Sensors does not exist
When you click drag another mobs sprite to pull up their equipment menu there is no option to adjust that individuals suit sensors.

### PR DESCRIPTION
**Pull request by Ziiro**
Adapted from Paradise.

Closes #299 

I'm not 100% on the code, but it works. It takes locked/no sensors into account, and broadcasts to everyone around that you're doing it.

Same interface

Alerts everyone in view
dreamseeker_2017-12-11_22-47-53
Takes locked sensors into account

:cl: Ziiro
add: You may now alter suit sensor levels of other people on their equip/un-equip menu.
/:cl:

### CODE REVIEW

File: code/modules/clothing/clothing.dm [hunk 1]
```diff
@@ -700,6 +700,20 @@ BLIND     // can't see anything
 				to_chat(usr, "<span class='notice'>Your suit will now only report your exact vital lifesigns.</span>")
 			if(3)
 				to_chat(usr, "<span class='notice'>Your suit will now report your exact vital lifesigns as well as your coordinate position.</span>")
+	else if(istype(src.loc, /mob))
+		switch(sensor_mode)
+			if(0)
+				for(var/mob/V in viewers(1, usr))
+					V.show_message("<span class='warning'>[usr] disables [src.loc]'s remote sensing equipment.</span>", 1)
```
Comment (SamHPurp) [inline code/modules/clothing/clothing.dm]: doublecheck, but I have a feeling all `[src.loc]` could and should just be `[loc]`

File: code/modules/mob/living/carbon/human/human.dm [hunk 2]
```diff
@@ -209,6 +209,9 @@ INITIALIZE_IMMEDIATE(/mob/living/carbon/human/dummy)
 		dat += "&nbsp;<A href='?src=\ref[src];pockets=right'>[(r_store && !(r_store.flags_1&ABSTRACT_1)) ? "Right (Full)" : "<font color=grey>Right (Empty)</font>"]</A></td></tr>"
 		dat += "<tr><td>&nbsp;&#8627;<B>ID:</B></td><td><A href='?src=\ref[src];item=[slot_wear_id]'>[(wear_id && !(wear_id.flags_1&ABSTRACT_1)) ? wear_id : "<font color=grey>Empty</font>"]</A></td></tr>"
 		dat += "<tr><td>&nbsp;&#8627;<B>PDA:</B></td><td><A href='?src=\ref[src];item=[slot_wear_pda]'>[(wear_pda && !(wear_pda.flags_1&ABSTRACT_1)) ? wear_pda : "<font color=grey>Empty</font>"]</A></td></tr>"
+		if(istype(w_uniform, /obj/item/clothing/under))
+			var/obj/item/clothing/under/U = w_uniform
+			dat += "<tr><td>&nbsp;&#8627;<B>Suit Sensors:</b></td><td><A href='?src=\ref[src];set_sensor=1'>[U.has_sensor >= 2 ? "<font color=grey>--SENSORS LOCKED--</font></a>" : "Set Sensors</a>"]</td></tr>"
```
Comment (SamHPurp) [inline code/modules/mob/living/carbon/human/human.dm]: Note to myself that if this merges first this'll conflict with my `\ref[src]` refactor

**Comment by SamHPurp** [comment]
Also note for Githubs auto-actions: This closes #299

**Comment by SamHPurp** [comment]
edited to include a changelog, as well as the Github hint to autoclose the related Issue

**Comment by FlattestGuitar** [comment]
Crimes, crimes, crimes.

Please refactor this to use `visible_message()`, not those loops.

**Comment by imsxz** [comment]
Would it make sense for suit sensors to take time to disable?

File: code/modules/clothing/clothing.dm [hunk 3]
```diff
@@ -700,6 +700,16 @@ BLIND     // can't see anything
...
+	else if(istype(loc, /mob))
+		switch(sensor_mode)
+			if(0)
+				usr.visible_message("<span class='warning'>[usr] disables [loc]'s remote sensing equipment.</span>")
+			if(1)
+				usr.visible_message("[usr] turns [loc]'s remote sensors to binary.")
+			if(2)
+				usr.visible_message("[usr] sets [loc]'s sensors to track vitals.")
+			if(3)
+				usr.visible_message("[usr] sets [loc]'s sensors to maximum.")
```
Comment (FlattestGuitar) [inline code/modules/clothing/clothing.dm]: All of the ones that still don't have a span should also be a `span class='notice'` pls.

Sorry for not having mentioned that earlier.

**Comment by Ziiro** [comment]
dreamseeker_2017-12-13_12-29-08

File: code/modules/mob/living/carbon/human/human.dm (hunk 2 above)
Comment (SamHPurp) [reply code/modules/mob/living/carbon/human/human.dm]: Welp, now you're going to need to update this ref yourself. Update from master to get the new defines...

This line should now be:

`dat += "<tr><td>&nbsp;&#8627;<B>Suit Sensors:</b></td><td><A href='?src=[REF(src)];set_sensor=1'>[U.has_sensor >= 2 ? "<font color=grey>--SENSORS LOCKED--</font></a>" : "Set Sensors</a>"]</td></tr>"`

**Pull request merged**